### ISSUE TITLE
Barcode Scanner's camera view temporarily messes up with translations

### ISSUE THREAD

**Issue by 2-4601** [WhisperSystems/Signal-Android#2898]
0. System language: English (US), TextSecure language: Japanese (or any translated language)
1. Go to a conversation
2. Tap identity verification from the lock icon
3. Choose to scan a code
4. After Barcode Scanner has opened, go back (don't have to scan anything)
5. Note that the title of this view has changed to English (also the overlay text from Barcode Scanner is in English "No scanned key found")
6. Go back to conversation view
7. All the menus are in English and also the text box

tl;dr: just watch this .gif:
![screenrecord2](https://cloud.githubusercontent.com/assets/174176/6995078/0b15ae26-db3a-11e4-8d38-95ed2265bb1d.gif)

If the system language is set to non-English there is no issue.
Also Barcode Scanner's QR code view does not cause this – only the camera.

Tested with: Android 5.0.1 and 4.0.4; TextSecure 2.10.1 (first time I run into this was in 2.6 or earlier)

**Comment by rhodey** [comment]
![](http://i.imgur.com/JSBTl.gif) fav bug report yet ^^

I added a debug statement [here](https://github.com/WhisperSystems/TextSecure/blob/master/src/org/thoughtcrime/securesms/util/DynamicLanguage.java#L23) and found that the `Locale` on `Context.getResources().getConfiguration().locale` changes from what is specified in TextSecure settings to the system locale in `DynamicLanguage.onResume()` after returning from the zxing barcode scanning intent.

**Comment by agrajaghh** [comment]
just a wild guess because there are some differences, but could this be related to #2815? I still cant reproduce it, but its happening sometimes...

**Comment by rhodey** [comment]
I'm a little confused about the difference between #2815 and #2684, but it seems that #2815 is about a bug with `DynamicLanguage` working during `onResume()`. If I'm understanding that right then I agree with @agrajaghh

**Comment by agrajaghh** [comment]
@rhodey yes, #2684 seems unrelated and should be fixed by your #2725 (didn't try it though)

**Comment by rhodey** [comment]
This bug is also reproducible on Android Gingerbread, just tested. From the discussion I've read online about Android Locale changing randomly, most people seemed to agree that it should be safe to check for system locale changes `onCreate()` and that we don't have to worry about it `onResume()`-- guess this might not be true?

**Comment by agrajaghh** [comment]
since @McLoo can reproduce #2815, he might try adding the check in `onResume()` and see if it works?

**Comment by McLoo** [comment]
Will try. 
Until then, u can try  to reproduc 2815 by rotating your phone. Funny language flip

**Comment by 2-4601** [comment]
@rhodey thanks :)

@McLoo Wow, rotating really causes the exact same change. I have disabled auto-rotate from system settings so I should never experience this. However _Barcode Scanner's_ camera view forces the phone to landscape mode anyway and when you return back to _TextSecure_ the screen rotates to portrait again and language changes to English.

**Comment by FeuRenard** [comment]
The external barcode scanner is not used anymore from 3.20.0 on.
So this can be closed.

**Comment by 2-4601** [comment]
Closing as essentially duplicate of #2815
